src/main/scala/diplomacy/Nodes.scala

```diff
@@ -10,6 +10,90 @@ import freechips.rocketchip.util.HeterogeneousBag
 import scala.collection.mutable.ListBuffer
 import scala.util.matching._
 
+/**
+ * Concepts, metaphors, and mnemonics to help with understanding Diplomacy code.
+ *
+ *
+ * # Abstract types
+ *
+ * Diplomacy is a set of abstractions for describing directed, acyclic graphs
+ * where parameters may be negotiated between nodes. These abstractions are
+ * expressed in the form of abstract classes, traits, and type parameters, which
+ * comprise nearly all of the types defined in this file.
+ *
+ * The NodeImp ("node implementation") is the main abstract type that determines
+ * the type parameters of all other abstract types. Defining a concrete
+ * implementation of NodeImp will therefore determine concrete types for all
+ * type parameters. For example, passing in a concrete instance of NodeImp to a
+ * SourceNode will fully determine concrete types for all of a SourceNode's type
+ * parameters.
+ *
+ * Specific applications of Diplomacy are expected to either extend these types
+ * or to specify concrete types for the type parameters. This allows for
+ * creating application-specific node, edge, parameter, and bundle types.
+ *
+ * Is an "edge" and a binding the same thing?
+ *
+ *
+ * # Inward/Outward vs. Upward/Downward
+ *
+ * Diplomacy defines two dimensions: inward/outward and upward/downward.
+ *
```

Comment:
So, is it correct to say that inward/outward refer to the way edges move, and upward/downward refer to the way parameters move? Is this a helpful summary?